Improve documentation for `liquid_alternative_*` (#12810)
minetest/minetest: doc/lua_api.txt

    * The cubic source node for a liquid.
    * Faces bordering to the same node are never rendered.
    * Connects to node specified in `liquid_alternative_flowing`.
    * You *must* set `liquid_alternative_source` to the node's own name.
    * Use `backface_culling = false` for the tiles you want to make
      visible when inside the node.
* `flowingliquid`
    * The flowing version of a liquid, appears with various heights and slopes.
    * Faces bordering to the same node are never rendered.
    * Connects to node specified in `liquid_alternative_source`.

>     * You *must* set `liquid_alternative_flowing` to the node's own name.

Reviewer: Wouldn't it then make sense to default this value to the node name directly? In which case would these differ?
Same question for drawtype `liquid` above.

        --              around it until `liquid_range` is reached;
        --              will drain out without a source;
        --              recommended drawtype: "flowingliquid".
        -- If it's "source" or "flowing", then the
        -- `liquid_alternative_*` fields _must_ be specified

        liquid_alternative_flowing = "",
        liquid_alternative_source = "",
        -- These fields may contain node names that represent the
        -- flowing version (`liquid_alternative_flowing`) and
        -- source version (`liquid_alternative_source`) of a liquid.
        --
        -- Specifically, these fields are required if any of these is true:
        -- * `liquidtype ~= "none" or
        -- * `drawtype == "liquid" or
        -- * `drawtype == "flowingliquid"
        --

>         -- Liquids can have up two nodes: source and flowing.

Reviewer: ```suggestion
        -- Liquids consist of up two nodes: source and flowing.
```

        --

>         -- There are two ways to define a liquid:
>         -- 1) with a source node and a flowing node
>         -- 2) with only a source node (when it never flows)
>         --
>         -- With option 1, you have to set both fields for both nodes.
>         -- With option 2, you have to set `liquid_alternative_source` and
>         -- `liquid_range` must be set to 0.

Reviewer: What do you think of placing the notes in-line?
```suggestion
        -- There are two ways to define a liquid:
        -- 1) Source node andflowing node. This requires both fields to be
        --    specified for both nodes.
        -- 2) Standalone source node (cannot flow). `liquid_alternative_source`
        --    and `liquid_range` must be set to 0.
```

        -- * `drawtype == "liquid" or
        -- * `drawtype == "flowingliquid"
        --

>         -- Liquids consist of up two nodes: source and flowing.

Reviewer 2: up to two nodes?
